extensions/replication/tasks/ReplicateObject.js

```diff
@@ -701,6 +748,14 @@ class ReplicateObject extends BackbeatTask {
                 this._setTargetAccountMd(destEntry, targetRole, log, next);
             },
             // Get data from source bucket and put it on the target bucket
+            next => {
+                if (!mdOnly &&
+                    sourceEntry.getContentLength() / 1000000 >=
+                    this.repConfig.queueProcessor.sourceCheckIfSizeGreaterThanMB) {
+                    return this._checkSourceReplication(sourceEntry, log, next);
+                }
+                return next();
```

Comment:
We should explicitly test that we only do the check for object larger than `sourceCheckIfSizeGreaterThanMB`